src/devices/acurite.c

```diff
@@ -745,6 +748,40 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
             valid++;
         }
 
+        // 515 sensor messages are 6 bytes.
+        if (browlen == ACURITE_515_BITLEN / 8) {
+        // Sensor type is determined by bit 0 from the message_type.
+        // 0 = refrigerator, 1 = freezer
+        sensor_type = message_type & 1 ? 'F' : 'R';
+
+            channel = acurite_getChannel(bb[0]);
+            // Sensor ID is the last 14 bits of byte 0 and 1
+            // CCII IIII | IIII IIII
+            // The sensor ID changes on each power-up of the sensor.
+            sensor_id = ((bb[0] & 0x3f) << 8) | bb[1];
+            // temperature encoding used by 515 sensors
+            // 14 bits available after removing both parity bits.
+            int temp_raw = ((bb[3] & 0x7F) << 7) | (bb[4] & 0x7F);
+            tempf = (temp_raw - 1480) * 0.1f;
+            sprintf(channel_str, "%c%c", channel, sensor_type);
+            // Battery status is the 7th bit 0x40. 1 = normal, 0 = low
+            battery_low = (bb[2] & 0x40) == 0;
+
+            /* clang-format off */
+            data = data_make(
+                    "model",                "",             DATA_STRING, _X("Acurite-515","Acurite 515 sensor"),
+                    "id",                   "",             DATA_INT,    sensor_id,
+                    "channel",              NULL,           DATA_STRING, &channel_str,
+                    _X("battery_ok","battery_low"), "",     DATA_INT,    _X(!battery_low,battery_low),
+                    "temperature_F",        "Temperature",  DATA_FORMAT, "%.1f F", DATA_DOUBLE, tempf,
+                    "mic",                  "Integrity",    DATA_STRING, "CHECKSUM",
```

Comment:
A check for the checksum is needed. Otherwise remove the mic indication.

Reply:
Ok, well the check is a few lines above this in the general case. Ignore my comment ;)

Reply:
It's checked further up. So when it gets here, it's a valid checksum.